index.d.ts

```diff
@@ -0,0 +1,81 @@
+export interface Action {
+  type: string;
+}
+
+
+/* reducers */
+
+export type Reducer<S> = <A extends Action>(state: S, action: A) => S;
+
+export function combineReducers<S>(reducers: {[key: string]: Reducer<any>}): Reducer<S>;
```

Comment:
I see one drawback here: until TypeScript supports [default type parameters](https://github.com/Microsoft/TypeScript/issues/2175) we'd have to always specify second parameter which adds verbosity.

How about overloads:
```ts
export function combineReducers<S>(reducers: ReducersMapObject): Reducer<S>;
export function combineReducers<S, M extends ReducersMapObject>(reducers: M): Reducer<S>;
```